dyninstAPI/src/emit-amdgpu.C

```diff
@@ -0,0 +1,573 @@
+  printf("emitLoadOrigRegRelative not implemented yet\n");
+}
+
+void EmitterAmdgpuGfx908::emitLoadOrigRegister(Address /* register_num */, Register /* dest */,
+                                               codeGen & /* gen */) {
+  printf("emitLoadOrigRegister not implemented yet\n");
+}
+
+void EmitterAmdgpuGfx908::emitStoreOrigRegister(Address /* register_num */, Register /* dest */,
+                                                codeGen & /* gen */) {
+  printf("emitStoreOrigRegister not implemented yet\n");
+}
+
+void EmitterAmdgpuGfx908::emitStore(Address /* addr */, Register /* src */, int /* size */,
+                                    codeGen & /* gen */) {
+  printf("emitStore not implemented yet\n");
+}
+
+void EmitterAmdgpuGfx908::emitStoreIndir(Register addr_reg, Register src, int size, codeGen &gen) {
+  emitStoreRelative(src, /*offset =*/0, addr_reg, size, gen);
+}
+
+void EmitterAmdgpuGfx908::emitStoreFrameRelative(Address /* offset */, Register /* src */,
+                                                 Register /* scratch */, int /* size */,
+                                                 codeGen & /* gen */) {
+  printf("emitStoreFrameRelative not implemented yet\n");
+}
+
+void EmitterAmdgpuGfx908::emitStoreRelative(Register source, Address offset, Register base,
+                                            int size, codeGen &gen) {
+  // Caller must ensure the following:
+  //
+  // 1. base is even aligned and base, base + 1 contain the address.
+  // 2. <size> registers starting from source hold the value to be stored.
+  // 3. offset must be a 21-bit signed value. Sign-extend to int64_t and cast to
+  // uint64_t before calling if necessary (this is bad).
+  // 3. Alignment requirement for source:
+  //     alignment = size
+  //     size = 1, 2, 4
+
+  assert(size == 1 || size == 2 || size == 4);
+
+  int alignment = size;
+  assert(source % alignment == 0 && "source register must be aligned");
+
+  assert(source + size - 1 <= SGPR_101 &&
+         "must have <size> consecutive registers from source to store <size> "
+         "words from");
+
+  unsigned storeOpcode = 0;
+  switch (size) {
+  case 1:
+    storeOpcode = S_STORE_DWORD;
+    break;
+  case 2:
+    storeOpcode = S_STORE_DWORDX2;
+    break;
+  case 4:
+    storeOpcode = S_STORE_DWORDX4;
+    break;
+  default:
+    assert(false && "size can only be 1, 2, or 4");
+  }
+
+  emitSmem(storeOpcode, source, (base >> 1), (uint64_t)offset, gen);
+}
+
+void EmitterAmdgpuGfx908::emitStoreShared(Register /* source */, const image_variable * /* var */,
+                                          bool /* is_local */, int /* size */,
+                                          codeGen & /* gen */) {
+  printf("emitStoreShared not implemented yet\n");
+}
+
+bool EmitterAmdgpuGfx908::emitMoveRegToReg(Register src, Register dest, codeGen &gen) {
+
+  emitSop1(S_MOV_B32, dest, src, /*hasLiteral =*/false, /*literal =*/0, gen);
+  return 0;
```

Comment:
For all bool returning functions use the constant `false` for a boolean value not `0`:
```suggestion
  return false;
```